pkg/manager/member/startscript/v2/pd_start_script.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"fmt"
+	"path/filepath"
+	"text/template"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/manager/member/constants"
+)
+
+// PDStartScriptModel contain fields for rendering PD start script
+type PDStartScriptModel struct {
+	PDDomain           string
+	PDName             string
+	DataDir            string
+	PeerURL            string
+	AdvertisePeerURL   string
+	ClientURL          string
+	AdvertiseClientURL string
+	DiscoveryAddr      string
+	ExtraArgs          string
+}
+
+// RenderPDStartScript renders PD start script from TidbCluster
+func RenderPDStartScript(tc *v1alpha1.TidbCluster) (string, error) {
+	m := &PDStartScriptModel{}
+	tcName := tc.Name
+	tcNS := tc.Namespace
+	peerServiceName := controller.PDPeerMemberName(tcName)
+
+	m.PDDomain = fmt.Sprintf("${PD_POD_NAME}.%s.%s.svc", peerServiceName, tcNS)
```

Comment:
Is it ok to remove `.svc` if no clusterDomain is specified to reduce the useless dns query?
And if we do this, the code in the discovery [here](https://github.com/pingcap/tidb-operator/blob/v1.3.2/pkg/discovery/discovery.go#L82) also needs to update.

Reply:
Please also check for the other components.

Reply:
OK, I see, this is to keep the backward compatibility with v1 start scripts.